pageserver/ctl/src/draw_timeline_dir.rs

```diff
@@ -83,10 +83,16 @@ fn parse_filename(name: &str) -> (Range<Key>, Range<Lsn>) {
     let keys: Vec<&str> = split[0].split('-').collect();
     let mut lsns: Vec<&str> = split[1].split('-').collect();
 
+    // Handle generation number
     if lsns.last().expect("should").len() == 8 {
         lsns.pop();
     }
 
+    // Handle version number
+    if lsns.last().expect("should").starts_with("v") {
+        lsns.pop();
+    }
+
```

Comment:
I'm surprised this works: the new style layer names have a "-v1-<generation>" suffix (see `local_layer_path` function), so isn't the generation still confusing this?  It might be misinterpreted as an LSN.

(if you feel like it, it might be worth just rewriting this to a regex, IDK why it was originally written as a bunch of splits and pops)

Reply:
L87 handles generation number and pops it out, L92 handles the -v1 suffix and pops it out, the remaining things are LSNs